planner/core/logical_plan_test.go

```diff
@@ -1670,6 +1670,53 @@ func (s *testPlanSuite) TestSimplyOuterJoinWithOnlyOuterExpr(c *C) {
 	c.Assert(join.JoinType, Equals, RightOuterJoin)
 }
 
+func (s *testPlanSuite) TestResolvingCorrelatedAggregate(c *C) {
```

Comment:
Should we also modify `checkDataSourceCols` like https://github.com/pingcap/tidb/pull/21431/files?file-filters%5B%5D=.go#diff-037390a673963744ffeb2623d0f17bc5205f52510dda89f3522d44309621f7fbR621?